### PR TITLE
Clarify policy windows

### LINKED ISSUE #16: policy window issues
I have a few questions / concerns about policy windows in the current draft:

1. Since policy windows are fixed per-issuer, that means that a single issuer cannot enforce policies of 3x/day and 10x/month at the same time. This seems to limit the flexibility of origins working with a given issuer. I'm not strongly opposed to this because in the current design you want to limit the policy diversity anyway. I guess my questions are a) is my understanding correct and b) should there be text about how an issuer chooses a policy window and therefore its impact on rate limitation policies?
2. How does an attester learn about the issuer policy window? Is the attester assumed to know the issuer configuration / discover it somehow? Apologies if I missed this.
3. The document doesn't have explicit recommendations for how long windows can be (I think a month is used in the example). Given this, client rekey penalization based on policy windows doesn't make sense to me. If the length is a month, it seems that a well-behaved client can certainly lose their keys twice within a two-month period (“a well-behaved Client will never change keys this frequently”); if the length is an hour, allowing the clients to rekey every two hours seems excessive
4. I think the text should be more explicit about the fact that client key rotation resets the client access tracking for the remaining policy window. Perhaps something like "If a client loses the client secret, the attester MAY allow the client to generate a new client key and secret and register the client key. However this will reset the client's policy window and thus could be used to exceed the rate limits, so an attester MUST penalize clients that register new keys too frequently"

### PR DESCRIPTION
**Pull request by tfpauly**
Closes #16

### CODE REVIEW

File: draft-ietf-privacypass-rate-limit-tokens.md [hunk 1]
```diff
@@ -232,7 +232,13 @@ error instead of the issuer's token response.
 ~~~
 {: #fig-example-deny title="Failed rate-limited issuance"}
 
-Each Issuer defines a window of time over which Attesters enforce rate limits.
+Each Issuer defines a window of time over which Attesters enforce rate limits,
+defined in the Issuer configuration directory ({{setup}}). This window might
+be as long as a month, or as short as an hour, depending on the use case. The
+window is the same across all Origins that work with the Issuer; if multiple
+window lengths are needed, then the entity running the Issuer can run multiple
+Issuers, one for each window length.
```
Comment (chris-wood) [inline draft-ietf-privacypass-rate-limit-tokens.md]:
```suggestion
Issuers, one for each window length, and each with a different name.
```

File: draft-ietf-privacypass-rate-limit-tokens.md [hunk 2]
```diff
@@ -576,7 +582,9 @@ to identify a Client.
 Attesters need to enforce that Clients don't change their Client Key frequently, to ensure Clients can't
 regularly evade the per-client policy as seen by the issuer. Attesters MUST NOT allow Clients to
 change their Client Key more than once within a policy window, or in the subsequent policy window
-after a previous Client Key change. Alternative schemes where the Attester stores the encrypted
+after a previous Client Key change. Chaning the secret will reset the client's policy window and
```
Comment (chris-wood) [inline draft-ietf-privacypass-rate-limit-tokens.md]:
```suggestion
after a previous Client Key change. Changing the secret will reset the client's policy window and
```

File: draft-ietf-privacypass-rate-limit-tokens.md [hunk 3]
```diff
@@ -576,7 +582,9 @@ to identify a Client.
...
+thus can be used to exceed rate limits, so an Attester MUST penalize clients that register new secrets
+too frequently. Alternative schemes where the Attester stores the encrypted
```
Comment (chris-wood) [inline draft-ietf-privacypass-rate-limit-tokens.md]:
```suggestion
thus can be used to exceed rate limits. One way to mitigate this is for the Attester to penalize clients that register new secrets
too frequently. Alternative schemes where the Attester stores the encrypted
```

**Review by chris-wood — APPROVED** [review]
Approved pending the normative language change around penalization, since it's we're not precise with its implementation. The suggested text simply offers it as a mitigation to the problem, which I do think is clearly described.

File: draft-ietf-privacypass-rate-limit-tokens.md (hunk 1 above)
Comment (tfpauly) [reply draft-ietf-privacypass-rate-limit-tokens.md]:
```suggestion
Issuers with different Issuer names, one for each window length.
```

**Pull request merged**